### PR TITLE
fix some transpose optimizer bugs

### PR DESCRIPTION
**Pull request by zhijxu-MS**
fix transpose optimizer bug

node can't be removed directly when graph output comes from it

### CODE REVIEW

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 1]
```diff
@@ -340,8 +340,13 @@ def _transpose_handler(self, trans, node):
             ops = self._g.get_nodes()
             self._g.replace_all_inputs(ops, node.output[0], trans.input[0])
 
+            shape = self._g.get_shape(node.output[0])
+            dtype = self._g.get_dtype(node.output[0])
```
Comment (nbcsm) [inline tf2onnx/optimizer/transpose_optimizer.py]: please add tests to cover these cases.

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 2]
```diff
@@ -380,7 +385,7 @@ def _mul_handler(self, trans, node):
                 multiplier_input_id = i
                 multiplier_input_node = input_node
 
-        if not multiplier_input_node.is_const():
+        if multiplier_input_node is None or not multiplier_input_node.is_const():
```
Comment (pengwa) [inline tf2onnx/optimizer/transpose_optimizer.py]: thanks for fixing that. add a comment for better understanding?

Comment (zhijxu-MS) [reply tf2onnx/optimizer/transpose_optimizer.py]: done

File: tf2onnx/optimizer/transpose_optimizer.py (hunk 1 above)
Comment (zhijxu-MS) [reply tf2onnx/optimizer/transpose_optimizer.py]: 1 transpose node is graph output has been covered by test_transpose_leaky_relu
2 identity is graph output is covered by new-added test case "test_transpose_with_identity"

**Pull request merged**